src/civilite/schedule.py

```diff
@@ -77,80 +144,31 @@ def __init__(self, tzinfo: pytz.tzinfo = pytz.utc, **kwargs) -> None:
-    if event.start > sunset_time:
-        result = EVT_FIXED
-    elif sunset_time > event.stop:
-        result = EVT_NEVER_ON
-    return result
-
-
-def createEvents(year: int, schedule: WeeklySchedule) -> Dict[date, Tuple[datetime, int, bool]]:
-    '''Create events from a Schedule object at a custom location for the entire given year.
-
-    Outputs the schedule to a CSV file.
-
-    Returns:
-        A dictionary defined as follows:
-            keys: dates of the year
-            values: tuple of (sunset datetime, event type, "event type changed" flag)
-
-    The "event type changed" flag indicates that a change in the clock's
-    manual programming is needed for the event that week.
-    '''
-    # Custom observer: Rochester_HoP,USA,43°09'N,77°23'W,US/Eastern,170
-    # Note: elevation (in meters) is just a guess based on ROC airport. We may adjust it.
-    observer = ScheduleObserver(latitude=43.1606355, longitude=-77.3883843, elevation=170,
-                                tzinfo=pytz.timezone("US/Eastern"))
-    data = {}
-    calendar_date = date(year, 1, 1)
-    event_type_changed = False
-    while calendar_date.year == year:
-        curr_event_type = getEventType(schedule, calendar_date, observer)
-        if curr_event_type is not None:
-            prev_week_event_type = getEventType(schedule,
-                                                calendar_date - timedelta(days=7),
-                                                observer)
-            if curr_event_type != prev_week_event_type:
-                event_type_changed = True
-        data[calendar_date] = (observer.getCivilTwilight(calendar_date),
-                               curr_event_type,
-                               event_type_changed)
-        calendar_date += timedelta(days=1)
-        event_type_changed = False
-    csv_name = f'sunsets_{year}.csv'
-    with open(csv_name, 'w') as out_file:
-        out_file.write('"Weekday","Date","Sunset","Event Type","Event Change?"\n')
-        for evt_date, (sunset_time, evt_type, evt_changed) in sorted(data.items()):
-            out_file.write(
-                f'"{calendar.day_abbr[evt_date.weekday()]}",'
-                f'{evt_date},'
-                f'{sunset_time.strftime("%H:%M:%S")},'
-                f'"{EVENT_TYPES.get(evt_type, "")}",'
-                f'{"*" if evt_changed else ""}\n'
-            )
-    print(f'Schedule {csv_name} created.')
-    return data
-
-
 def getCurrentSchedule() -> WeeklySchedule:
     '''Current HoP weekly schedule.'''
-    result = WeeklySchedule()
+    # Observer at HoP: Rochester_HoP,USA,43°09'N,77°23'W,US/Eastern,170
+    # Note: elevation (in meters) is just a guess based on ROC airport. We may adjust it.
+    hop_observer = ScheduleObserver(latitude=43.1606355, longitude=-77.3883843, elevation=170,
+                                    tzinfo=pytz.timezone("US/Eastern"))
+    result = WeeklySchedule(hop_observer)
```

Comment:
Technically, we don't need the hop_observer object... so you could clean this up to create the object in the constructor of WeeklySchedule. 
    result = WeeklySchedule( ScheduleObserver(latitude=43.1606355, longitude=-77.3883843, elevation=170,
                                    tzinfo=pytz.timezone("US/Eastern")))

Reply:
Actually, I'm trying to get hop_observer entirely out of code and into configuration instead so that an Application client can instantiate it out there and pass it in.  That way the library will become more useful in general.